packages/eslint-plugin/src/rules/prefer-readonly.ts

```diff
@@ -232,7 +236,13 @@ export default util.createRule<Options, MessageIds>({
           classScopeStack[classScopeStack.length - 1].enterNonConstructor();
         }
       },
-      [`${functionScopeBoundaries}:exit`](node: TSESTree.Node): void {
+      [`${functionScopeBoundaries}:exit`](
```

Comment:
NGL - I hate the fact that `:exit` is a special case selector which only needs to be at the end.

It's so confusing to me that `'ArrowFunctionExpression, FunctionDeclaration, FunctionExpression, MethodDefinition:exit'` means "select any of these node types, and run the function on any of their exits"

I kind of expected you to need to do `:matches( ... ):exit`